src/plots/linePlot.ts

```diff
@@ -349,10 +373,59 @@ export module Plots {
 
     protected _getDataToDraw() {
       let dataToDraw = new Utils.Map<Dataset, any[]> ();
-      this.datasets().forEach((dataset) => dataToDraw.set(dataset, [dataset.data()]));
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        if (!this._croppedRendering) {
+          dataToDraw.set(dataset, [data]);
+          return;
+        }
+
+        let filteredDataIndices = data.map((d, i) => i);
```

Comment:
sorry, now that I think more about it, this is needed in order to merge croppedRendering and downSampling